controllers/platformnetwork_controller.go

```diff
@@ -1325,3 +1333,43 @@ func (r *PlatformNetworkReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.PlatformNetwork{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *PlatformNetworkReconciler) checkRestoreInProgress(instance *starlingxv1.PlatformNetwork) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status and clear annoatation
+func (r *PlatformNetworkReconciler) RestorePlatformNetworkStatus(instance *starlingxv1.PlatformNetwork) error {
+	r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Status without reconcilation: %t", instance.Status.InSync)
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	if annotation != nil {
+		config, ok := annotation[cloudManager.RestoreInProgress]
+		if ok {
+			platformNetworkStatus := &starlingxv1.PlatformNetworkStatus{}
+			err := json.Unmarshal([]byte(config), &platformNetworkStatus)
+			if err == nil {
+				instance.Status.InSync = platformNetworkStatus.InSync
+				instance.Status.Reconciled = platformNetworkStatus.Reconciled
+				instance.Status.DeploymentScope = platformNetworkStatus.DeploymentScope
```

Comment:
What if annotation doesn't provide correct values?
Suggest not to set values to `instance.Status.*` unless values are specified and correct.
Can we have following new struct in manager.go:
```
type RestoreStatus struct {
	InSync          *bool   `json:"inSync"`
	Reconciled      *bool   `json:"reconciled"`
	DeploymentScope *string `json:"deploymentScope"`
}
```

and use the struct `cloudManager.RestoreStatus` to unmarshal JSON across all the controllers?

1. Update `instance.Status.InSync` only when `RestoreStatus.InSync != nil`
2. Update  `instance.Status.Reconciled` only when `RestoreStatus.Reconciled != nil`
3. Update  `instance.Status.DeploymentScope` only when `RestoreStatus.DeploymentScope != nil && (strings.ToLower(*RestoreStatus.DeploymentScope) == "bootstrap" || strings.ToLower(*RestoreStatus.DeploymentScope) == "principal" )

I believe this would make the code more robust.

Reply:
Added RestoreStatus struct and modified the code accordingly.